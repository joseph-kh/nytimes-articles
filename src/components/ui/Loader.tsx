import React from "react";
import {
  StyleSheet,
  ActivityIndicator as ActivityIndicatorElement,
  ActivityIndicatorProps,
} from "react-native";
import Center from "@/components/ui/containers/Center";
import COLORS from "@/theme/colors";

const Loader: React.FC<ActivityIndicatorProps> = (props) => {
  const { size = "large", color = COLORS.primary } = props;

  return (
    <Center style={styles.container}>
      <ActivityIndicatorElement {...props} size={size} color={color} />
    </Center>
  );
};

export default Loader;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
