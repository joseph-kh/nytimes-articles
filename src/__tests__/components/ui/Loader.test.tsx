import React from "react";
import { create } from "react-test-renderer";
import Loader from "@/components/ui/Loader";

describe("Loader Component", () => {
  it("renders correctly", () => {
    const tree = create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const customStyle = { backgroundColor: "blue" };
    const tree = create(<Loader style={customStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
