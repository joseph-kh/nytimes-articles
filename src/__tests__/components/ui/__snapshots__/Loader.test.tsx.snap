// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component applies custom styles 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#51e2c3"
    size="large"
    style={
      {
        "backgroundColor": "blue",
      }
    }
  />
</View>
`;

exports[`Loader Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#51e2c3"
    size="large"
  />
</View>
`;
